### DISCUSSION TITLE
Controlling Outputs on DDRC-24 does not work

### DISCUSSION THREAD

**Discussion by d45p00k13** [mrene/minidsp-rs#574] [Q&A] [answered]
Hey everybody!
I want to start with a great thank you for spending so much effort on software that will help me very much, so thanks again!

I have a miniDSP DDRC-24 with Dirac Live enabled for my subwoofer (in/out 1) and for my shaker (in/out 2). I want to use ezBEQ to set the appropriate filters for the movies I wanna watch and there want to use the underlying minidsp-rs executable (on Windows 11) which sends its commands to a miniDSP WI-DG. The problem I will describe is valid for USB connection also.

I can switch the master volume, the configs and such using the command line. I was wondering why the sound does not change at all. The SW shows no failure but when I want to control the outputs to configure PEQs (what is the main focus because of ezBEQ of course) or just the individual gain, nothing happens and is applied to the outputs (reviewed with the miniDSP software).

The util just shows no problem but nothing happens. Here is the command I used that worked:
".\minidsp.exe --tcp=192.168.5.5:5333" adding "config 1" or "... gain -- -30"

what does not work:
adding "output 0 gain -- -10" for example.

Since I did not find any examples on how to set the output PEQ using the commands (the help does not present any idea on how the syntax is for the set commands at all).

Since the connection works fine, it looks like a problem with the compatibility with the integration of DDRC-24, I seek help to get everything working. Besides, the ezBEQ command line / debug output does not recognize any problems either but no PEQ is set when looking at the miniDSP software (what should be the case after reading the application manual from miniDSP themselves for minidsp-rs).

Best regards,
Spookster

**Comment by mrene** [comment]
Hi!

I also have a DDRC-24 with Dirac (but no WI-DG, I only use USB connectivity), and I can correctly change the individual output gain with `minidsp output 0 gain -- -30`.

These devices have a limitation in that it is not possible to read their current configuration. It is therefore expected that when making changes to input and output gains, that these changes are not reflected in the official software, because this software relies on its previous information and has no mechanism to read the current device's configuration.

Can you add `-vvv` to your minidsp command?  `.\minidsp.exe -vvv --tcp=192.168.5.5:5333" output 0 gain -- -30`  - it will output the commands that get sent to your device and how it responds.

**Comment by d45p00k13** [comment]
Hey,

thanks for the fast response. Here is the data! 

PS  .\minidsp.exe -vvv --tcp=192.168.5.5:5333 output 0 gain -- -10
[0.000s] Sent: [31]
[0.001s] Sent: ReadHardwareId
[0.006s] Recv: [31, 01, 32, 0a]
[0.006s] Recv: HardwareId(HardwareId { fw_major: 01, fw_minor: 32, hw_id: 0a })
[0.007s] Sent: [05, ff, a1, 01]
[0.008s] Sent: ReadMemory { addr: ffa1, size: 01 }
[0.011s] Recv: [05, ff, a1, 65]
[0.011s] Recv: MemoryData(MemoryView { base: ffa1, data: [65] })
[0.012s] Sent: [05, ff, fe, 02]
[0.013s] Sent: ReadMemory { addr: fffe, size: 02 }
[0.016s] Recv: [05, ff, fe, 0c, 85]
[0.017s] Recv: MemoryData(MemoryView { base: fffe, data: [0c, 85] })
[0.018s] Sent: [13, 80, 00, 22, 00, 00, 20, c1]
[0.019s] Sent: Write { addr: Addr { val: 22, len: 03, extra_bit: true }, value: Value { Bytes: [0, 0, 20, c1] (i: 0 | i32: 8385 | f32: -10.0 | fp: FixedPoint(0.0009995699)) } } (0x22 | 34) <> D_GAIN_1_0
[0.023s] Recv: []
[0.023s] Recv: Ack

Since I can see the change of the master volume in the original software, I thought this would be the case for the outputs also... looks like a crappy SW Design from miniDSP :S is there a way to check if everything is in place somehow? Maybe with some additional information about the syntax to control the "set" command, I could set a specific PEQ I can compare with the original software and play specific sounds like pink noise to check if the frequencies are boosted. What do you think?

**Reply by mrene** [reply, marked as answer]
This looks like line, the last "write" and "ack" means the device has accepted the change.

For validation, I'm comparing the command sent by the application when setting a PEQ, [and checking that it's the same data](https://github.com/mrene/minidsp-rs/blob/094e8717b306c7e2e239f26059b4bb0495d557c4/minidsp/tests/m2x4hd.rs#L65-L78). 

You could do a sanity check and set an obvious filter to see what changes. I'm surprised that a -30dB output gain wasn't obvious though, maybe you can set it to -127dB and observe that no sound is coming out of your subwoofer?

For PEQs, the syntax for the command line requires setting the coefficients directly (either on the command, or by importing a file which you can save from the minidsp app, or from REW, or other softwares). There is a [website](https://arachnoid.com/BiQuadDesigner/index.html) where you can design filters, and then you can do `minidsp output 0 peq 0 set b0 b1 b2 a1 a2`, while replacing the appropriate values.

For example, this will set a -16dB low shelf filter at 500hz cutting removing all the bass, `minidsp output 0 peq 0 set -- 0.9564662817448897 -1.8565420093653364 0.9016605752914146 1.8523345768114219 -0.8623342895902185` which will be very obvious to your ears (use `minidsp output 0 peq 0 clear` to revert it)

**Reply by d45p00k13** [reply]
Yeah... i did not see the wood for the trees^^ setting the overall output gain to -127 works as intended. Then obviously i need to check my other settings like the limiter and such that might level out the additional boost from eqBEQ. Thank you so much for enlighten me and being now a happy user of minidsp-rs. Great work and thanks for the support!
